sdk/core/azure-core/README.md

```diff
@@ -2,21 +2,51 @@
 
-The library allows client libraries to expose common functionality in a consistent fashion.  Once you learn how to use these APIs in one client library, you will know how to use them in other client libraries.
+The library allows client libraries to expose common functionality in a consistent fashion. Once you learn how to use these APIs in one client library, you will know how to use them in other client libraries.
 
 ## Getting started
 
-Typically, you will not need to download `azure-core`; it will be downloaded for you as a dependency of the client libraries.  In case you want to download it explicitly (to implement your own client library, for example), you can find the source
-in here.
+Typically, you will not need to download `azure-core`; it will be downloaded for you as a dependency of the client libraries. In case you want to download it explicitly (to implement your own client library, for example), you can find the source in here, or use vcpkg to install the package `azure-core-cpp`.
 
 ## Key concepts
 
-The main shared concepts of Azure::Core include:
-- Configuring service cliesnt, e.g. configuring retries, logging, etc.. (`ClientOptions`)
-- Accessing HTTP response details (`Response`, `Response<T>`)
-- Polling long-running operations
-- Exceptions for reporting errors from service requests in a consistent fashion (`RequestFailedException`)
-- Abstractions for Azure SDK Credentials (`TokenCredential`)
+The main shared concepts of `Azure::Core` include:
+
+- HTTP pipeline and HTTP policies such as retry and logging, which are configurable via service client specific options.
+- Handling streaming data and input/output (I/O) via `BodyStream` along with its derived types.
+- Accessing HTTP response details from the method return types, via `Response<T>`.
+- Polling long-running operations (LROs), via `Operation<T>`.
+- Exceptions for reporting errors from service requests in a consistent fashion via the base exception type `RequestFailedException`.
+- Abstractions for Azure SDK credentials (`TokenCredential`).
+- Replaceable HTTP transport layer to send requests and receive responses over the network.
+
+### Long Running Operations
+
+Some operations take a long time to complete and require polling for their status. Methods starting long-running operations return `Operation<T>` types.
+
+You can intermittently poll whether the operation has finished by using the `Poll()` method on the returned `Operation<T>` and track progress of the operation using `Value()`. Alternatively, if you just want to wait until the operation completes, you can use `PollUntilDone()`.
+
+```C++
+SomeServiceClient client;
+
+auto operation = *client.StartSomeLongRunningOperation();
+
+while (!operation.IsDone())
+{ 
+  std::unique_ptr<Http::RawResponse> response = operation.Poll();
+
+  auto partialResult = operation.Value();
```

Comment:
Well, this depends completely on the LRO derived class.
a partial result is not something that an operation<T> can ensure.  This would require prior knowledge about the specific LRO derived class providing this capability.

This might mislead a customer expecting partialResult to be something offer by Operation<T>.

A simpler approach is to loop forever until isDone is true or some timeout occur
```cpp
while(true) {
  auto rawHttpResponse = operation.Poll();
  if (operaion.IsDone) {
  // It is guaranteed that the operation<T> will set isDone = true if last Poll() was the last one
  break;
  }
 
  // if you want to log, you need to use the rawHttpResponse for it. In this case, you should have previous knowledge about how does the response would look like

  slepp(...);
```

Reply:
That's fine if all LROs don't support this. This is more to showcase conceptual patterns for some that do. I want to highlight more of the `Operation<T>` APIs.

> This might mislead a customer expecting partialResult to be something offer by Operation.

This conceptual doc is showing patterns, and not guaranteeing all operations support partial. But some do, so we'd want to highlight that.

Reply:
Yes, in this case, <T> = GetBlobPropertiesResult, and that class acts as 2 things:
- CopyResult
- CopyProgress

The progress reporting capability in this case should be advertised by `StartCopyBlobResult` class and not from `Operation<T>`.  There is no contract on Operation<T> for this feature.